cni.go

```diff
@@ -310,3 +320,13 @@ func (c *libcni) GetConfig() *ConfigResult {
 func (c *libcni) reset() {
 	c.networks = nil
 }
+
+func (c *libcni) ready() error {
```

Comment:
we are not consistently "unlocked" when calling ready() here.. need to have two ready funcs..

first one calls the second one.. second one does not "lock" and does the work..

Thus any verb that already has the lock calls the second one directly.. verbs that are not in the lock call the one that puts them in the lock.

Reply:
or, of course we could pull the lock out of ready and just make sure all verbs that need ready are in the lock..

Reply:
or fix golang, and make their locks reentrant :-O